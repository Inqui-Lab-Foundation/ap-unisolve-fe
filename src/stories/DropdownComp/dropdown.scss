.studentDropdown{
    .dropdownMenuButton1{
        background: #FFCB34 !important;
        border-radius: 60px !important;
    }
    
}