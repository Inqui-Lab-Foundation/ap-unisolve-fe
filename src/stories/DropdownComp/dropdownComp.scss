.studentDropdown {
  select {
    option {
      background-color: red !important;
    }
  }
  .single-dropdown {
    border: 0.1rem solid var(--color-grey-light-3);
    box-sizing: border-box;
    border-radius: 6rem;
    background: var(--color-white-1);
    font-size: 1.6rem;
    color: var(--color-black-2);
    width: 100%;
    text-align: left;
    padding: 1.5rem 1.6rem;

    // appearance: none;
    // -webkit-appearance: none;
    // -moz-appearance: none;
    // -ms-appearance: none;
    // background: url("../../media/arrown-down.png") no-repeat !important ;
    // background-position: right 1.5rem center !important;
    // background-size: 1rem;
    // background-size: 1.5rem !important;

    &:focus,
    &:focus-visible {
      opacity: 0.6;
      border: 0.2rem solid var(--color-blue);
      box-shadow: none;
      outline: none;
    }
  }
  .defaultDropdown {
    border-radius: 1rem !important;
    // padding: 1rem 1rem ;
  }
}
