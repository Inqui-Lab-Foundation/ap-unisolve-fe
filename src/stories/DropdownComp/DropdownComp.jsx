import React, { useState } from "react";
import "./dropdown.scss";
import PropTypes from "prop-types";

export const DropDownComp = ({
  options,
  backgroundColor,
  size,
  label,
  onChange,
  onBlur,
  values,
  name,
  ...props
}) => {
  return (
    <div className="dropdown studentDropdown">
      <select
        id={name}
        name={name}
        onChange={onChange}
        onBlur={onBlur}
        className="single-dropdown"
      >
        <option value="">Select</option>
        {options.map((item,i) => {
          return <option key={i} value={item}>{item}</option>;
        })}
      </select>
    </div>
  );
};
DropDownComp.propTypes = {
  /**
   * Is this the principal call to action on the page?
   */
  // SingleSelectDropdown: PropTypes.bool,
  /**
   * What background color to use
   */
  // backgroundColor: PropTypes.string,
  /**
   * How large should the button be?
   */
  size: PropTypes.oneOf(["small", "medium", "large"]),
  /**
   * Button contents
   */
  label: PropTypes.string.isRequired,
  /**
   * Optional click handler
   */
  onClick: PropTypes.func,
};
DropDownComp.defaultProps = {
  backgroundColor: null,
  size: "medium",
  onClick: undefined,
  label: "Dropdown",
  options: [
    "Argentina",
    "Austria",
    "Cocos Islands",
    "Kuwait",
    "Sweden",
    "Venezuela",
    "city",
  ],
};
