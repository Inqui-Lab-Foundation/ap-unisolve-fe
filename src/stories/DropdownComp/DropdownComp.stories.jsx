import React from 'react';

import { DropDownComp } from './DropdownComp';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
    title: 'Example/DropDownComp',
    component: DropDownComp,
    // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
    argTypes: {
        backgroundColor: { control: 'color' },
    },
};

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template = (args) => <DropDownComp {...args} />;

export const SingleSelectDropdown = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
SingleSelectDropdown.args = {
    primary: true,
    label: 'Select Class',
    options:["Garde 1","Garde 2","Garde 3","Garde 1","Garde 2","Garde 3"]

};

export const DefaultDropdown = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
DefaultDropdown.args = {
    primary: true,
    label: 'Select Class',
    options:["Garde 1","Garde 2","Garde 3","Garde 1","Garde 2","Garde 3"],
    className:"defaultDropdown"
};