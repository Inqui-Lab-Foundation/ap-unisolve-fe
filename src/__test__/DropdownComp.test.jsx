import React from "react";
import { shallow,configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {DropDownComp} from '../stories/DropdownComp/DropdownComp';

configure({adapter: new Adapter()})

describe('MyComponent',() =>{
    it('Should show text',() =>{shallow
        const wrapper= (<DropDownComp />);
        const text = wrapper.find('option');
        expect(text).toMatchSnapshot();
    })
})