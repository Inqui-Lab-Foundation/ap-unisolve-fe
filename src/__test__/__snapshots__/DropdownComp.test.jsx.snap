// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent Should show text 1`] = `ShallowWrapper {}`;
